CONTRIBUTING.md

```diff
@@ -335,6 +335,31 @@ since the latest release over the hotfixed `master`.
 This would mean, however, that the commits referenced from PRs previously merged to `develop` will no longer be part of `develop`'s history,
 which is rather unacceptable.
 
+### Plugin signing
+
+The valid non-expired key pair together with the password required for the plugin signing should be present in the CI environment.
+If they are absent, please take a look at the [plugin signing in IntelliJ](https://plugins.jetbrains.com/docs/intellij/plugin-signing.html#signing-methods) for updated instructions on how to do it.
+Currently, you would need to first generate the private key with:
+```shell
+openssl genpkey\
+  -aes-256-cbc\
+  -algorithm RSA\
+  -out private.pem\
+  -pkeyopt rsa_keygen_bits:4096\
+  -pass env:PASSWORD_ENVIRONMENT_VARIABLE_NAME
+```
+Then, use that for generating the certificate chain by running:
+```shell
+openssl req\
+  -key private.pem\
+  -new\
+  -x509\
+  -days 365\
```

Comment:
Actually if you could check what happens if you use an expired certificate... will you be able to:
1. sign the plugin using an expired cert?
2. install a plugin signed using an expired cert?

Reply:
This is how I am going to test it:
- created a cert which expires tomorrow.
- today I would sign with it, tomorrow will check if it can be installed
- tomorrow, I would use the expired cert to sign the plugin, to see if it is allowed.

Reply:
>  just sign the plugin without publishing

would check if it is possible. But if we merge this PR. each time `publishPlugin` is getting run locally, it would fail.

Reply:
I'd never run `publishPlugin` locally by design, anyway ;) publishing should be restricted to CI

Sooo if it turns out there's no task to **only sign** in Gradle, pls also include the IntelliJ CLI command in the docs (again, for the sake of local testing)

Reply:
there is `signPlugin` 👍

Reply:
Wow nice 🎉 so pls use in the sample for local usage instead of the signer CLI

Reply:
But it does not work properly. I had not set the env vars in the terminal session. It did produce the signed zip and said `Build Successful`

Reply:
after cleaning the `distributions` directory. and running `publishPlugin` again to produce a zip, it does not even sign anymore.

Reply:
> It did produce the signed zip and said Build Successful

Huh I'll check later today, it might qualify for an issue in gradle-intellij-plugin (that wouldn't be [the first time](https://github.com/JetBrains/gradle-intellij-plugin/issues?q=is%3Aissue+is%3Aclosed+author%3APawelLipski) :trollface:)

Reply:
> I had not set the env vars in the terminal session. It did produce the signed zip and said Build Successful

Huh that's somewhat weird... could you check the signatures on the zip generated in this way?

> and running publishPlugin again to produce a zip, it does not even sign anymore.

Does `publishPlugin` task depend on `signPlugin` by default? is there a need to introduce an explicit dependency?

Reply:
in the output for `./gradlew assemble --dry-run publishPlugin`
we have the following lines in order, regardless of whether the plugin signing is configured or not:
```
:signPlugin SKIPPED
:verifyPlugin SKIPPED
:publishPlugin SKIPPED
```

Reply:
Huh okay... but isn't it just an expected behavior of `--dry-run` flag? 🤔

Reply:
in another run: when the env vars are not set, 
- the `publishPlugin`task fails, but still produces the zip file.
- the `signPlugin` task succeeds, but produces no zip file.

Reply:
https://github.com/JetBrains/gradle-intellij-plugin/blob/76a8e9301f73df66160d6ddcd308cbeab05fc084/src/main/kotlin/org/jetbrains/intellij/IntelliJPlugin.kt#L1304

Reply:
so `publishPlugin` is explicitly configured to depend on the `signPlugin` task.
The way we can detect if the signing was successful or not, is through the name suffix of the distribution's zip file, if it is suffixed with `signed`, this means that the signing task was successful.

Reply:
Okay... so the last thing before we can proceed with merging is for your to check the issues around cert expiry ☝🏻

Reply:
It also, both allows signing with an expired plugin, and installing a plugin with an expired certificate.

Reply:
`openssl x509 -enddate -noout -in <(echo "$PLUGIN_SIGN_CERT_CHAIN") -checkend 604800` would output the certificate expiry. Maybe I can add a script to condition the build to the validity of the cert?

Reply:
Hmmm let's leave as is for now... approving the PR then, I'll generate the production key/cert soon and fill up the CI secrets